### DISCUSSION TITLE
WooCommerce REST API: Is Timing Out During Batch Product Creation

### DISCUSSION THREAD

**Discussion by refaatcrafts** [woocommerce/woocommerce#52060] [General]
Hello WooComm Community
I'm trying to use the WooCommerce REST API to batch create "push" products, but I'm encountering an issue. It seems to successfully push between 2 and 6 products before timing out. Is this normal behavior, or could it be related to using a free WordPress provider like Pantheon  Any insights or suggestions would be greatly appreciated! Thanks

**Comment by barryhughes** [comment]
Hi @nerdxio,

https://woocommerce.com/document/server-requirements/

It would definitely be worth reviewing the above guidance, and determining if your host meets those specs. Beyond that, a lot depends on the nature of the payload you are sending. Images, in particular, can result in significant slowdowns if you supply a large number of URLs, but it's hard to know what the specific issue is in your case without more information.
